### PR TITLE
Add tests for CUDA mean and norm.

### PR DESCRIPTION
**Pull request by sriharshakandala**
Add tests for CUDA mean and norm.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try-

**Comment by sriharshakandala** [comment]
bors try

**Review by simonbyrne — APPROVED** [review]

File: test/Fields/reduction_cuda.jl [hunk 1]
```diff
@@ -93,5 +95,24 @@ end
     # test minimum
     @test local_min[1] ≈ FT(0)
     @test local_min[1] ≈ local_min_cpu
+    # testing mean
+    meanz = Statistics.mean(Z)
+    meanz_cpu = Statistics.mean(Z_cpu)
+    @test meanz[] ≈ meanz_cpu[]
```
Comment (simonbyrne) [inline test/Fields/reduction_cuda.jl:101]: doesn't this return a scalar? why do you need `[]`?

Comment (sriharshakandala) [reply test/Fields/reduction_cuda.jl:101]: The cuda version returns a Vector of size `Nf`.

**Pull request merged**